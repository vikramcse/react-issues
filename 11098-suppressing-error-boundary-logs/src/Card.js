import React from "react";

const Card = props => {
  const initials = `${props.item.name.split(" ")[0][0]}.${
    props.item.name.split(" ")[1][0]
  }`;
  return (
    <li>
      <img
        src={props.item.image}
        alt={props.item.name}
        height="100"
        width="100"
      />
      <h3>{props.item.name}</h3> ({initials})
      <p>{props.item.role}</p>
    </li>
  );
};

export default Card;
