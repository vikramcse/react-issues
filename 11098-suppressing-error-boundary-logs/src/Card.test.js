import React from "react";
import ReactTestUtils from "react-dom/test-utils";

import Card from "./Card";
// "patrick Jane"
test("Card", () => {
  let item = {
    name: 1,
    role: "Consultant",
    image: "https://upload.wikimedia.org/wikipedia/en/3/37/Patrick_Jane.jpg"
  };
  let index = 1;

  try {
    const component = ReactTestUtils.renderIntoDocument(
      <Card item={item} index={index} />
    );
  } catch (e) {}
});
