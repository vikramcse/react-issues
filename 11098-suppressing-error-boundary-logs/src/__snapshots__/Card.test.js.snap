// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card 1`] = `
<li>
  <img
    alt="patrick Jane"
    height="100"
    src="https://upload.wikimedia.org/wikipedia/en/3/37/Patrick_Jane.jpg"
    width="100"
  />
  <h3>
    patrick Jane
  </h3>
   (
  p.J
  )
  <p>
    Consultant
  </p>
</li>
`;
